MaeBlock.hpp

```diff
@@ -294,6 +294,14 @@ template <typename T> class IndexedProperty
         return operator[](index);
     }
 
+    inline const T& at(size_type index, const T& default_) const
+    {
+        if (m_is_null && (*m_is_null)[index]) {
```

Comment:
OK, I understand what this means now...the name m_is_null is pretty terrible for readability (ahem @mbeachy :) ), as if the pointer itself is null, then it's false, if the pointer is true it's true...and you can't tell the type from the name to distinguish what it really is (in this line you get the hint later that it's a pointer to a vector or array pointer).

Anyway, end rant.  LGTM.